Code/Java/CircularQueue

```diff
@@ -0,0 +1,360 @@
+  The pointers to the first and last element are dynamic. 
+                                
+           First element -> O - O                   Last element -> O - O <- First element
+                           /     \                                 /     \
+           Last element -> O     O      may also be                O     O
+                           \     /                                 \     /
+                            O - O                                   O - O 
+  
+  
+        O - O - O - O - O - O           may also be         O - O - O - O - O - O
+        ^                   ^                               ^   ^
+       first               last                           last first
+  
+  In a normal Queue, we can insert elements until queue becomes full. But once queue becomes full, 
+  we can not insert the next element even if there is a space in front of queue. This is possible
+  in Circular Queues and hence is a optimizes space consumption.
+ */
+
+import java.util.*;
+public class CircularQueue
+{
+    int cq[];
+    int first, last;
+
+    CircularQueue(int size)
+    { 
+        //This is a default constructor to declare Circular Queue and global parameters first and last.
+        cq = new int[size];
+        first =-1;
+        last = -1;
+    }
+
+    public void insert(int value)
+    {
+         // This function is used to insert an element into the circular queue. 
+         // In a circular queue, the new element is always inserted at last position. 
+         
+        if(first == 0 && last == cq.length-1 || first == last+1)
+        { 
+            System.out.println("Overflow"); return;
+        }
+        else if(first == -1)
+        {
+            first=last=0;
+        }
+        else if(last == cq.length-1)
+        {
+            last=0;
+        }
+        else
+        {
+            last++;
+        }
+        cq[last]=value;
+    }
+
+    public int delete()
+    {
+        // This function is used to delete an element from the Circular Queue. 
+        // In a circular queue, the element is always deleted from first position. 
+         
+        if(first == -1)
+        {
+            System.out.println("Underflow"); return 0; 
+        }
+        else
+        {
+            int temp = cq[first];
+            if(first == last)
+            { 
+                reset(); 
+            }
+            else if(first == cq.length-1)
+            {
+                first=0;
+            }
+            else
+            {
+                first++;
+            }
+            return temp;
+        }
+    }
+
+    private void display()
+    {
+        // This function is used to display a Circular Queue.
+         
+        if(first == -1)
+        {
+            System.out.print("Circular Queue is Empty");
+            return;
+        }
+        
+        for(int i = first; ; i++)
+        {
+            if(i == cq.length)
+            {
+                i=0;
```

Comment:
```suggestion
                i = 0;
```

Add space around the operators like =, <, > etc

Reply:
Changes made.